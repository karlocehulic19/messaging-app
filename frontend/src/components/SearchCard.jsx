import PropTypes from "prop-types";
import { useState, useEffect } from "react";
import customFetch from "../utils/customFetch";

export default function SearchCard({ username, photoId = null }) {
  const [imgSrc, setImgSrc] = useState();

  useEffect(() => {
    customFetch(`/profile-picture/${photoId}`)
      .then((res) => res.blob())
      .then((img) => {
        const profPic = URL.createObjectURL(img);
        setImgSrc(profPic);
      });
  }, [photoId]);

  return (
    <div aria-label={`${username} user`}>
      <img src={imgSrc} alt={`${username} profile picture`} />
      <span>{username}</span>
    </div>
  );
}

SearchCard.propTypes = {
  username: PropTypes.string.isRequired,
  photoId: PropTypes.string,
};
