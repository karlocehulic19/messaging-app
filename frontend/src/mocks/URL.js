global.URL.createObjectURL = vi.fn(() => "mockedFilePath");
global.URL.revokeObjectURL = vi.fn(() => null);
