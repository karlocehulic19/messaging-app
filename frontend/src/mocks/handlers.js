import { http, HttpResponse } from "msw";
import { config } from "../Constants";

export const handlers = [
  http.post(`${config.url.BACKEND_URL}/login`, async ({ request }) => {
    const body = await request.json();
    if (!body.username || !body.password) {
      return new HttpResponse("Missing credentials", { status: 401 });
    }

    if (body.username != "someUsername" || body.password != "somePassword") {
      return HttpResponse.json(
        { messages: ["Username or password is incorrect"] },
        { status: 401 }
      );
    }
    return HttpResponse.json({ token: "randomJWTtoken" }, { status: 200 });
  }),
];
