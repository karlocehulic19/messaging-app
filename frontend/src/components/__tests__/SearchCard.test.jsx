import { describe, expect, vi } from "vitest";
import { screen, render, waitFor } from "@testing-library/react";
import SearchCard from "../SearchCard";
import { server } from "../../mocks/node";

server.listen();

global.URL.createObjectURL = vi.fn(() => {
  return "mockedDataURL";
});

describe("<SearchCard />", () => {
  it("renders correctly", async () => {
    render(<SearchCard username="Test" photoId="testid1" />);

    await waitFor(() => expect(global.URL.createObjectURL).toBeCalledTimes(1));

    expect(screen.getByLabelText("Test user")).toMatchSnapshot();
  });
});
