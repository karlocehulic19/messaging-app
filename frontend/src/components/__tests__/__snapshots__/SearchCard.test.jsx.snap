// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`<SearchCard /> > renders correctly 1`] = `
<div
  aria-label="Test user"
>
  <img
    alt="Test profile picture"
    src="mockedDataURL"
  />
  <span>
    Test
  </span>
</div>
`;
